plog/plog

```diff
@@ -459,6 +469,28 @@ main () {
     shift $(($OPTIND - 1))
 
 
+# Check compression type
+    if $(command -v  bzip2 >/dev/null) ; then
+        PLOG_COMPRESSED_EXT="${PLOG_COMPRESSED_EXT:-bz2}"
+    else
+        PLOG_COMPRESSED_EXT="${PLOG_COMPRESSED_EXT:-gz}"
+    fi
+
+    if [[ ! "$PLOG_COMPRESSED_EXT" =~ ^(bz2|gz)$ ]]; then
+        echo "PLOG_COMPRESSED_EXT must be bz2 or gz, was $PLOG_COMPRESSED_EXT"
+        exit 1
+     fi
+
+     if [[ "$PLOG_COMPRESSED_EXT" =~ bz2 ]]; then
+         compress=bzip2
+         uncompress=bunzip2
+         zegrep=bzegrep
+     else
+         compress=gzip
+         uncompress=gunzip
+         zegrep=zegrep
+     fi
```

Comment:
I'm thiking this should go in the `compress` and `uncompress` functions because that's were all the compress logic is already but also I think it's a bit confusing having functions and variables with the same name and similar functionality. (I had to check they don't share a namespace!)

Reply:
I guess I was just lucky with the namespaces. :) The next push will fix this, mostly.

Reply:
Now cut-down to just the egrep program to use